src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryOperators.cs

```diff
@@ -562,7 +675,38 @@ private static FormulaValue TimeDifference(IRContext irContext, FormulaValue[] a
             var arg1 = (TimeValue)args[1];
 
             var result = arg0.Value.Subtract(arg1.Value);
-            return new TimeValue(irContext, result);
+            return new NumberValue(irContext, result.TotalDays);
+        }
+
+        private static FormulaValue SubtractDateAndTime(IRContext irContext, FormulaValue[] args)
+        {
+            DateTime arg0;
+            switch (args[0])
+            {
+                case DateTimeValue dtv:
+                    arg0 = dtv.Value;
+                    break;
+                case DateValue dv:
+                    arg0 = dv.Value;
+                    break;
+                default:
+                    return CommonErrors.RuntimeTypeMismatch(irContext);
+            }
+
+            var arg1 = (TimeValue)args[1];
+
+            var result = arg0.Subtract(arg1.Value);
+            return new DateTimeValue(irContext, result);
+        }
+
+        private static FormulaValue SubtractNumberAndDate(IRContext irContext, FormulaValue[] args)
```

Comment:
>SubtractNumberAndDate [](http://example.com/codeflow?start=36&length=21)

This operation (and the next one) are currently accepted by the compiler/binder. But their results are wrong (both in C# and in JS). So we'll keep them working in the compiler (otherwise it would be a breaking change for Power Apps), but always return an error.